### PR TITLE
[expo-modules-core] fix prebuilding error

### PR DESCRIPTION
**Pull request by Kudo**
# Why

fix `et prebuild-packages expo-modules-core` error caused by react-native header dependencies.

# How

add custom modulemap and include React-Core header while building expo-modules-core.

i was once worried for the generated xcframework would be large and accidentally include react-native symbols. fortunately, it doesn't. the generated xcframework is 5.8MB, compared to UMCore it's larger where i think coming from swift runtime. verifying the symbols by `nm -C ExpoModulesCore.xcframework/ios-arm64/ExpoModulesCore.framework/ExpoModulesCore`

# Test Plan

`et prebuild-packages expo-modules-core` successfully to generate xcframework.
please note that there was a xcode 12.5 crash issue for building expo-modules-core xcframework. please test with xcode 13. related issue: https://bugs.swift.org/browse/SR-14541 (thanks @tsapeta for sharing this information)

### CODE REVIEW

File: packages/expo-modules-core/ios/ExpoModulesCore.h [hunk 1]
```diff
@@ -1,5 +1,71 @@
...
+#import <ExpoModulesCore/EXTaskServiceInterface.h>
+#import <ExpoModulesCore/EXUIManager.h>
+#import <ExpoModulesCore/EXUnimodulesCompat.h>
+#import <ExpoModulesCore/EXUserNotificationCenterProxyInterface.h>
+#import <ExpoModulesCore/EXUtilities.h>
+#import <ExpoModulesCore/EXUtilitiesInterface.h>
+#import <ExpoModulesCore/EXViewManager.h>
+#import <ExpoModulesCore/EXViewManagerAdapter.h>
+#import <ExpoModulesCore/EXViewManagerAdapterClassesRegistry.h>
+#import <ExpoModulesCore/ExpoModulesCore.h>
```
Comment (tsapeta) [inline packages/expo-modules-core/ios/ExpoModulesCore.h:71]: Are you sure we need all of this? I think the same headers are already imported by `*-umbrella.h` 🤔 
Also, importing `ExpoModulesCore.h` seems to be circular 😅

File: tools/src/prebuilds/XcodeGen.ts [hunk 2]
```diff
@@ -130,6 +130,7 @@ export async function createSpecFromPodspecAsync(
 
         // Suppresses deprecation warnings coming from frameworks like OpenGLES.
         VALIDATE_WORKSPACE_SKIPPED_SDK_FRAMEWORKS: arrayize(podspec.frameworks).join(' '),
+        MODULEMAP_FILE: '$(SRCROOT)/module.modulemap',
```
Comment (tsapeta) [inline tools/src/prebuilds/XcodeGen.ts:133]: We should probably check if the file exists before we set it here 🤔

**Comment by Kudo** [comment]
@tsapeta the previous work was just a demonstration 😅. please review again for the final version.

File: tools/src/prebuilds/Prebuilder.ts [hunk 3]
```diff
@@ -160,7 +162,12 @@ export async function buildFrameworksForProjectAsync(
 export async function cleanTemporaryFilesAsync(xcodeProject: XcodeProject) {
   logger.log('   Cleaning up temporary files');
 
-  const pathsToRemove = [`${xcodeProject.name}.xcodeproj`, INFO_PLIST_FILENAME];
+  const pathsToRemove = [
+    `${xcodeProject.name}.xcodeproj`,
+    INFO_PLIST_FILENAME,
+    'generated.modulemap',
```
Comment (tsapeta) [inline tools/src/prebuilds/Prebuilder.ts:168]: Make it a constant since it's used twice 😉

File: tools/src/prebuilds/XcodeGen.ts [hunk 4]
```diff
@@ -212,3 +213,44 @@ function podNameToBundleId(podName: string): string {
     .replace(/(\_|[^\w\d\.])+/g, '.')
     .replace(/\.*([A-Z]+)/g, (_, p1) => `.${p1.toLowerCase()}`);
 }
+
+/**
+ * Generate custom modulemap for expo-modules-core where requiring React-Core as modular headers
```
Comment (tsapeta) [inline tools/src/prebuilds/XcodeGen.ts:218]:
```suggestion
 * Generate custom modulemap for expo-modules-core which needs to make React-Core headers modular
```

File: tools/src/prebuilds/XcodeGen.ts [hunk 5]
```diff
@@ -212,3 +213,44 @@ function podNameToBundleId(podName: string): string {
...
+ */
+export async function generateExpoModulesCoreModulemapAsync(destDir: string): Promise<string> {
+  const expoModulesCoreUmbrellaHeader = path.join(
+    PODS_PUBLIC_HEADERS_DIR,
+    'ExpoModulesCore',
+    'ExpoModulesCore-umbrella.h'
```
Comment (tsapeta) [inline tools/src/prebuilds/XcodeGen.ts:224]: It's also used more than once, so make it a global constant or a constant just here in this function 😉

**Review by tsapeta — APPROVED** [review]
Now it looks great, good job! 💪 Thanks for fixing it 🙏

In the future we will need to change it a little bit, because we will have to do the same thing with dev client's packages, but it's good for now 😄

**Review by lukmccall — APPROVED** [review]
Awesome 🔥  Looks great 🏅

**Pull request merged**